### PR TITLE
DR-1208 bulk load performance tuning

### PR DESCRIPTION
**Pull request by ddietterich**
Three changes:
1. Improved driver performance by only waving if there is nothing to do.
2. Change application properties to allow for many more stairway threads and many more concurrent file loads
3. Fix a bug in migrate where it was leaving cached connection set to read only

### CODE REVIEW

File: src/main/java/bio/terra/service/upgrade/Migrate.java [hunk 1]
```diff
@@ -199,16 +204,20 @@ private void releaseDeploymentLock() {
         try (Connection connection = dataSource.getConnection();
              PreparedStatement statement = connection.prepareStatement(updateSql)) {
             startTransaction(connection);
-            DeploymentRow row = getDeploymentRow(connection);
-            if ((row.getLockingPodName() == null) && (migrateConfiguration.getDropAllOnStart())) {
-                // Application of the migration released the lock for us. Nothing for us to do.
-                return;
-            }
+            try {
```
Comment (fboulnois) [inline src/main/java/bio/terra/service/upgrade/Migrate.java:223]: The triple-nesting of the `try ... catch` logic makes it a bit difficult to reason about this section. Can a level of nesting be avoided here? Return the result of `commitReadOnlyTransaction` and then have something like:

```
} catch (SQLException ex) {
  [ ... ]
} finally {
  if(startedTransaction) {
    commitReadOnlyTransaction(connection);
  }
}
```

**Review by fboulnois — CHANGES_REQUESTED** [review]
One suggestion:

File: src/main/java/bio/terra/service/filedata/flight/ingest/IngestDriverStep.java [hunk 2]
```diff
@@ -143,11 +143,13 @@ public StepResult undoStep(FlightContext context) {
     private void waitForAny(FlightContext context, UUID loadId, int concurrentLoads, int originallyRunning)
         throws DatabaseOperationException, InterruptedException {
         while (true) {
-            waiting();
+            // Is evaluating the  state of the load candidates is not instant,
+            // we recheck right away before waiting only wait if we did not find any new completions3
```
Comment (RoriCremer) [inline src/main/java/bio/terra/service/filedata/flight/ingest/IngestDriverStep.java]: I'm having a hard time understanding this comment

File: src/main/java/bio/terra/service/upgrade/Migrate.java (hunk 1 above)
Comment (ddietterich) [reply src/main/java/bio/terra/service/upgrade/Migrate.java:223]: Yeah, the problem is that `connection` is out of scope in the finally clause.
I split the inner part of the loop into his separate method and added the comment. See if that helps.

File: src/main/java/bio/terra/service/filedata/flight/ingest/IngestDriverStep.java (hunk 2 above)
Comment (ddietterich) [reply src/main/java/bio/terra/service/filedata/flight/ingest/IngestDriverStep.java]: Grumble! This voice dictation is a real pita

**Review by snf2ye — APPROVED** [review]
I took a detailed look at this PR earlier this week. It still looks good to me!

**Review by fboulnois — APPROVED** [review]
latest changes lgtm!

**Pull request merged**